finetune/adapter_v2.py

```diff
@@ -103,10 +103,13 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path):
     add_adapter_v2_parameters_to_linear_layers(model)
     mark_only_adapter_v2_as_trainable(model)
 
-    num_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
+    trainable_params = [p for p in model.parameters() if p.requires_grad]
+    num_params = sum(p.numel() for p in trainable_params)
     fabric.print(f"Number of trainable parameters: {num_params}")
+    num_params = sum(p.numel() for p in model.parameters() if not p.requires_grad)
+    fabric.print(f"Number of non trainable parameters: {num_params}")
 
-    optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
```

Comment:
Passing `trainable_params` doesn't have an impact. Added for explicitness.

`foreach=False` could be used here, but the `max_seq_length` memory spike dominates the memory usage. Meaning that we don't really need it in the fine-tuning scripts (see spikes in https://github.com/Lightning-AI/lit-gpt/pull/178#discussion_r1236180675). If set, per-iter times go increase from ~225ms to ~260ms so I didn't add it.

However, it is useful when using the regular non-finetuning model (as in the pretaining scripts) because the spike is not dominating in that case. See below:

`foreach=True`
![image](https://github.com/Lightning-AI/lit-gpt/assets/11067892/9d0b0636-6fe5-4c55-8782-ec98b46842e2)

`foreach=False`
![image](https://github.com/Lightning-AI/lit-gpt/assets/11067892/b90249c1-6824-4f59-9fca-aa5ddacac1c8)